cocos/core/director.ts

```diff
@@ -44,6 +44,8 @@ import { errorID, error, assertID, warnID } from './platform/debug';
 import { containerManager } from './memop/container-manager';
 import { uiRendererManager } from '../2d/framework/ui-renderer-manager';
 import { deviceManager } from '../gfx';
+import { PipelineBuilder } from '../rendering/custom/pipeline';
+import { macro } from './platform/macro';
```

Comment:
`interface PipelineBuilder` is used to build pipeline when custom pipeline is enabled.
Pipeline name is provided by `macro`.